src/Microsoft.AspNetCore.SystemWebAdapters/Adapters/SystemWebAdapterModule.Framework.cs

```diff
@@ -2,24 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Collections.Generic;
 using System.Web;
+using Microsoft.Extensions.DependencyInjection;
 
 namespace Microsoft.AspNetCore.SystemWebAdapters;
 
 internal sealed class SystemWebAdapterModule : IHttpModule
```

Comment:
I was actually going to do it that way originally, but it seems that the module is created multiple times for different requests which doesn't work for our scenarios since the session service needs to store state (locked session cache) between requests. I'll read up on IHttpModule lifetime and see if there's something we can do to change that behavior, but just from some ad-hoc testing I found the provider couldn't live here because it resulted in different requests having different locked session caches.

Reply:
oh good point - I don't think we can know how long the module itself is created. we probably don't wan it to dispose of anything then here and we handle that elsewhere (unless we're using a scoped instance in the module that it can own).

Reply:
Now that you mention it, this raises some more questions about service lifetime. We're registering IHttpModules as singleton currently but based on my observation that SystemWebAdapterModule might be instantiated multiple times, we could end up calling Init multiple times on the same registered modules which probably isn't desirable. 

Because SystemWebAdapterModule is the logical place to initialize the other modules, it may make sense to register modules with scope lifetime (though services can still be singleton for sharing caches, etc.) and create a scope in SystemWebAdapterModule's init that is disposed when that module is disposed.